examples/llama2/llama2_auto_opt.json

```diff
@@ -0,0 +1,89 @@
+    "input_model": {
+        "type": "HfModel",
+        "model_path": "meta-llama/Llama-2-7b-hf",
+        "generative": true,
+        "load_kwargs": { "attn_implementation": "eager" }
+    },
+    "data_configs": [
+        {
+            "name": "wikitext2_train",
+            "type": "HuggingfaceContainer",
+            "load_dataset_config": { "data_name": "wikitext", "subset": "wikitext-2-raw-v1", "split": "train" },
+            "pre_process_data_config": { "add_special_tokens": false, "max_samples": 128 }
+        },
+        {
+            "name": "gqa_transformer_prompt_dummy_data",
+            "type": "TransformersPromptDummyDataContainer",
+            "load_dataset_config": { "use_fp16": true, "shared_kv": true }
+        },
+        {
+            "name": "gqa_transformer_token_dummy_data",
+            "type": "TransformersTokenDummyDataContainer",
+            "load_dataset_config": { "use_fp16": true, "shared_kv": true }
+        },
+        { "name": "transformer_token_dummy_data", "type": "TransformersTokenDummyDataContainer" },
+        { "name": "transformer_prompt_dummy_data", "type": "TransformersPromptDummyDataContainer" }
+    ],
+    "systems": {
+        "local_system": {
+            "type": "LocalSystem",
+            "accelerators": [ { "accelerator_type": "gpu", "execution_providers": [ "CPUExecutionProvider" ] } ]
+        }
+    },
+    "evaluators": {
+        "merged_evaluator": {
+            "metrics": [
+                {
+                    "name": "latency_prompt_processing",
+                    "type": "latency",
+                    "sub_types": [ { "name": "avg", "priority": 1 } ],
+                    "data_config": "transformer_prompt_dummy_data",
+                    "user_config": { "io_bind": true }
+                },
+                {
+                    "name": "latency_token_generation",
+                    "type": "latency",
+                    "sub_types": [ { "name": "avg" } ],
+                    "data_config": "transformer_token_dummy_data",
+                    "user_config": { "io_bind": true }
+                }
+            ]
+        },
+        "gqa_evaluator": {
+            "metrics": [
+                {
+                    "name": "latency_prompt_processing",
+                    "type": "latency",
+                    "sub_types": [ { "name": "avg", "priority": 1 } ],
+                    "data_config": "gqa_transformer_prompt_dummy_data",
+                    "user_config": { "io_bind": true, "shared_kv_buffer": true }
+                },
+                {
+                    "name": "latency_token_generation",
+                    "type": "latency",
+                    "sub_types": [ { "name": "avg" } ],
+                    "data_config": "gqa_transformer_token_dummy_data",
+                    "user_config": { "io_bind": true, "shared_kv_buffer": true }
+                }
+            ]
+        }
+    },
+    "auto_optimizer_config": {
+        "enable": true,
```

Comment:
What would be the best behavior here? If auto opt config exists, workflow should run auto opt regardless passes in the config.

Reply:
It's carry-forward from previous implementation (used to be called `disable_auto_optimizer`).

Yes, this flag is redundant in current implementation. Existence of `auto_optimizer_config` should be enough to decide whether or not to run auto optimizer.

Flag removed.